modules/QueueManager.js

```diff
@@ -52,33 +52,19 @@ exports.addAllFair = videos => {
 
 //Add a video to the first position of the queue
 exports.addToTop = video => {
-    // Found a double! Move the video to the top
-    if(findDoppelganger(video)) queue.splice(queue.indexOf(doppelganger)).unshift(video);
-    //Video is not already in the queue, so add it to the top
-    else queue.unshift(video);
+    const doppelganger = findDoppelganger(video);
+
+    // Found a double! Remove the video from the queue
+    if(doppelganger) queue.splice(queue.indexOf(doppelganger), 1);
+    //Video is not in the queue, so add it to the top (prefer to insert original video again)
+    queue.unshift(doppelganger ?? video);
 }
 
 //Update the current video with the video in queue position 0, and remove it from the queue
-exports.moveToNext = async () => {
+exports.moveToNext = () => {
     // Queue has an item, can be shifted
-    if(queue.length > 0){
-        currentVideo = queue.shift()
-        if(currentVideo.user.user_id) {
-            try {
-                await fetch(`${process.env.LARAVEL_ENDPOINT}/api/protube/played?` + new URLSearchParams({
-                    secret: process.env.LARAVEL_API_KEY,
-                    user_id: currentVideo.user.user_id,
-                    video_id: currentVideo.id,
-                    video_title: currentVideo.title
-                }));
-            } catch(e) {
-                //TODO add nice error
-                console.log("Failed to send video data to protube site");
-            }
-        };
-    }else {
-        throw new softError('Unable to move to the next item in the queue!');
-    }
```

Comment:
It isn't removed but moved to the playbackmanager instead. Instead of being called at the start of the queue it is now done when the video has finished playing. This because when you switch from a video to the radio and back it'll have otherwise added the video twice. But I do see that I am missing the fetch import statement in the playbackmanager, will change that

Reply:
Ah, I'm blind. Good work!